metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py

```diff
@@ -583,36 +671,53 @@ def _process_table(
             self.report.report_dropped(table_identifier)
             return
 
-        table.columns = self.get_columns_for_table(
-            conn, table.name, schema_name, db_name
-        )
-        table.pk = self.get_pk_constraints_for_table(
-            conn, table.name, schema_name, db_name
-        )
-        table.foreign_keys = self.get_fk_constraints_for_table(
-            conn, table.name, schema_name, db_name
-        )
+        try:
+            table.columns = self.get_columns_for_table(
+                conn, table.name, schema_name, db_name
+            )
+        except Exception as e:
+            self.report_warning(
+                "columns-for-table",
+                f"Failed to get columns for table {table_identifier} due to error {e}",
+            )
+
+        try:
+            table.pk = self.get_pk_constraints_for_table(
+                conn, table.name, schema_name, db_name
+            )
+        except Exception as e:
+            self.report_warning(
+                "keys-for-table",
+                f"Failed to get primary key for table {table_identifier} due to error {e}",
+            )
+
+        try:
```

Comment:
Code Style: I'd like to break this down into more readable methods if posible

Reply:
done. Also refractored some other large methods.